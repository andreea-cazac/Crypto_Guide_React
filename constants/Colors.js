export const Colors = {
// Original (Your current favorite)
  style: {
    primary: '#14213D',    // Oxford Blue
    secondary: '#AEC6CF',  // Pastel Blue-Grey
  },
};
