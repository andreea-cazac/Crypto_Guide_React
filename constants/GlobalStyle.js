const primaryColor = '#14213D';
const secondaryColor = '#AFD5F0';
const white = '#fff';
const gray = '#555';
const errorColor = '#cc0000';
const successColor = '#00aa00';

export const GlobalStyle = {
  colors: {
    primary: primaryColor,
    secondary: secondaryColor,
    background: white,
    errorColor: errorColor,
    successColor: successColor,
    subtleText: gray,
  },

  components: {
    title: {
      fontSize: 30,
      fontWeight: 'bold',
      marginVertical: 16,
      color: primaryColor,
    },
  },
};