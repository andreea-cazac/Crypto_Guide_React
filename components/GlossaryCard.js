import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Colors } from '../constants/Colors';

const GlossaryCard = ({ term, meaning }) => {
    return (
        <View style={styles.card}>
            <Text style={styles.term}>{term}</Text>
            <Text style={styles.meaning}>{meaning}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    card: {
        borderRadius: 12,
        padding: 16,
        marginBottom: 16,
        backgroundColor: '#f5f5f5'
    },
    term: {
        fontSize: 18,
        fontWeight: 'bold',
        color: Colors.style.primary,
        marginBottom: 6,
    },
    meaning: {
        fontSize: 14,
        color: '#333',
        lineHeight: 20,
    },
});

export default GlossaryCard;