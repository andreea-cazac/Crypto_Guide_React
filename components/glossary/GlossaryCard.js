import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {GlobalStyle} from '../../constants/GlobalStyle';

const GlossaryCard = ({ term, meaning }) => {
    return (
        <View style={styles.card}>
            <Text style={styles.term}>{term}</Text>
            <Text style={styles.meaning}>{meaning}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    card: {
        borderRadius: 12,
        padding: 16,
        marginBottom: 16,
        backgroundColor: GlobalStyle.colors.background
    },
    term: {
        fontSize: 18,
        fontWeight: 'bold',
        color: GlobalStyle.colors.primary,
        marginBottom: 6,
    },
    meaning: {
        fontSize: 14,
        color: GlobalStyle.colors.primary,
        lineHeight: 20,
    },
});

export default GlossaryCard;