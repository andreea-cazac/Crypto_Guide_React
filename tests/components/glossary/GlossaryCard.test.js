// __tests__/GlossaryCard.test.js
import React from 'react';
import { render } from '@testing-library/react-native';
import GlossaryCard from '../../../components/glossary/GlossaryCard';
import { GlobalStyle } from '../../../constants/GlobalStyle';

describe('GlossaryCard', () => {
    const mockTerm = 'Test Term';
    const mockMeaning = 'This is a test meaning for the term.';

    it('renders the term and meaning correctly', () => {
        const { getByText } = render(
            <GlossaryCard term={mockTerm} meaning={mockMeaning} />
        );

        // Verify that the term text is rendered.
        expect(getByText(mockTerm)).toBeTruthy();
        // Verify that the meaning text is rendered.
        expect(getByText(mockMeaning)).toBeTruthy();
    });

    it('matches the snapshot', () => {
        const tree = render(
            <GlossaryCard term={mockTerm} meaning={mockMeaning} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});