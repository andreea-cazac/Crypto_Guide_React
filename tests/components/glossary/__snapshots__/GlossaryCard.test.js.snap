// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlossaryCard matches the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 12,
      "marginBottom": 16,
      "padding": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#14213D",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 6,
      }
    }
  >
    Test Term
  </Text>
  <Text
    style={
      {
        "color": "#14213D",
        "fontSize": 14,
        "lineHeight": 20,
      }
    }
  >
    This is a test meaning for the term.
  </Text>
</View>
`;
